### ISSUE TITLE
Fusion/Laya: eleven open PRs — merge order, what the measurements changed, and three decisions that need you

### ISSUE THREAD

**Issue by 402goose** [hathbanger/orc#26]
Eleven PRs are open on this repo from `402goose`, all against the Laya decision layer you landed in `0f9f039`. They were built and measured over one session against a real local Laya install, not just unit-tested. This issue is the executable handoff: merge order, what each one does, what the measurements changed about the plan, and the three decisions only a human can make.

Nothing here needs to be taken on trust — every claim below links to the comment with the numbers.

---

## TL;DR for the impatient

- Ten PRs are mergeable and verified **together**; one (#22) is a draft I recommend closing, with evidence.
- The single most important thing: **shadow mode does not currently collect on real workflows** (#20). Two bugs in the runtime mean one failed decision silences every later decision in a run, and explicit-lane workflows trip it every time. Until #20 lands, the Laya integration records almost nothing on a normal run.
- The measurements changed the plan: **`review` specialty is the strongest question in the system, and `recovery` cannot be made to work** on this checkpoint. Don't have anyone spend a month labeling recovery decisions.

---

## 1. Merge them

Everything except #22 merges onto `main` in this order. Conflicts are only both-sides additions — keep both sides in every case.

```sh
git checkout main
for b in fusion-shadow-collects fusion-policy-hardening fusion-acceptance-decision \
         fusion-calibration-metrics fusion-decision-benchmark fusion-coherence-checks \
         fusion-telemetry-deploy-docs fusion-telemetry-summary \
         fusion-cache-hit-telemetry fusion-codex-usage-mapping; do
  git merge --no-edit "402goose/$b"
done
```

| step | branch | PR | conflicts |
|---|---|---|---|
| 1 | `fusion-shadow-collects` | #20 | none |
| 2 | `fusion-policy-hardening` | #21 | `test/decisions_test.py` (both add tests) |
| 3 | `fusion-acceptance-decision` | #19 | `test/decisions_test.py`, `FUSION_DECISIONS.md` Verification section |
| 4 | `fusion-calibration-metrics` | #23 | none |
| 5 | `fusion-decision-benchmark` | #24 | `FUSION_DECISIONS.md` Verification section |
| 6 | `fusion-coherence-checks` | #25 | none |
| 7–10 | the four telemetry branches | #14 #15 #16 #17 | none |

**One manual tidy afterwards:** #19 and #24 each add a `laya_smoke.py` line to the Verification section, so you end up with both `--train` and `--train --acceptance` listed. Keep the latter.

**If you'd rather merge once:** `402goose/orc:mergeable-stack` is exactly the above, already resolved.

**Verify:**

```sh
make test          # 104 tests, must end in a bare OK
make dogfood
make dogfood-paired   # needs `orc fusion decisions setup` first; ~45s
cd telemetry && TEST_DATABASE_URL='postgres://localhost/orc_telemetry_test?sslmode=disable' go test ./...
```

All four pass on the combined tree. `make test` failing with "did not end in a bare OK" means a test skipped — that guard is #25 and it is deliberate.

---

## 2. What each PR does

**#20 — `fix(fusion): make shadow mode actually collect on real workflows`. Merge this first; the rest is less useful without it.**
Three bugs stopping shadow mode from producing data:
- `LayaRuntime` treated a per-request error as a dead process and latched it, so one failed decision made every later decision in the run report the same error at 0 ms.
- `route_task()` sent a one-option `choice` for every explicit lane, which crashes Laya (`torch.topk(2)` in `laya/common.py:122`). Combined with the latch: on any explicit-lane workflow, routing failed and then nothing else ran.
- Gate outcomes never reached telemetry. `trace_span()` runs inside `dispatch()` at `fusion_core.py:987`; `_accept_node` runs later at `fusion_workflow.py:800`, so a node the gate marks `invalid` was recorded as `success` locally and in the collector. Adds one gate span per acceptance decision — no schema change on the sender or the Go ingest.
Also: `workflow report` now prints each node's recorded recommendations and a `Gate: N accepted, M rejected` line, and `make dogfood-paired` asserts shadow mode changes no outcome.
[Live before/after](https://github.com/hathbanger/orc/pull/20#issuecomment-5767094224) · [flake follow-up](https://github.com/hathbanger/orc/pull/20#issuecomment-5767526329)

**#21 — `fix(fusion): shared quota vocabulary, shared cooldown constant, stop on repeated identical failures`.**
`_quota_failure` (lane cooldown) and `failure_class` (recovery classification, telemetry) had drifted: only the former knew `credits` and `too many requests`, and only the former read the summary. A 402 or 429 paused the node correctly while its recovery decision was logged as `worker_error` — and that record is what the learning export trains on. Also unifies the 900s cooldown constant, and stops a node whose blockers exactly repeat the previous attempt's.

**#19 — `feat(fusion): semantic acceptance Done-check as a fifth Laya decision kind`.**
Closes the last self-reported axis: `turn.failed`/`is_error` already make crash-vs-refusal structural, and `_accept_node` proves files changed and commands exited 0, but a clean exit-0 "success" was still taken verbatim from the worker's `STATUS:` line. Adds an `acceptance` kind that runs only after the structural checks pass and can only add a blocker, never accept.
Worth reading both comments — **the question wording was measured twice and wrong twice before it was right**: a double-barreled phrasing [inverted the live model](https://github.com/hathbanger/orc/pull/19#issuecomment-5766334035) (legitimate success scored *implausible* 0.81), then a paired-agreement rule [turned out to block the one case the check exists for](https://github.com/hathbanger/orc/pull/19#issuecomment-5767194966).

**#23 — `feat(fusion): calibration ECE, reliability bins, certain-error counts, evaluate --control`.**
`accuracy` and `selective_accuracy` cannot tell a model that says 0.9 and is right 90% of the time from one that says 0.9 and is right 60% — and the active-mode gate is "hard-branch on calibrated probability". Adds ECE with a ten-bin reliability table and confident-error counts at 0.9/0.95/0.99. `evaluate --control` scores held-out examples against a *different* example's state; on the stock checkpoint accuracy 1.0 vs control 0.0, i.e. it does read the state.

**#24 — `test(fusion): benchmark the checkpoint against Fusion's own decision schemas`.** See §3 — this one changes the plan.

**#25 — `test(fusion): guard the closed vocabularies that live in more than one file`.**
Adding a decision kind touches four places (`KINDS`, the `auto_actions` error text, the CLI's `--kind` choices, its questions map); miss the last and `probe --kind newkind` raises `KeyError`. A node status outside every terminal set makes `_final_status()` report `running` forever. Five checks that derive from the source of truth. Each was verified by injecting the real mistake, not just by passing.

**#14 #15 #16 #17 — telemetry.** #15 is the aggregate readout (`fusion telemetry report`) the whole group trial depends on; land it with #20. [Compatibility re-verified](https://github.com/hathbanger/orc/pull/15#issuecomment-5767490570).

---

## 3. What the measurements changed about the plan

`test/laya_benchmark.py` (#24) runs authored cases with deliberate near-misses against whatever checkpoint is installed. On the stock English one:

```
review (specialty)     4/4 clear-cut correct   p=0.62-0.70   distinct answers=4/4
intake (workflow)      5/6 clear-cut correct   p=0.44-0.96   distinct answers=4/8
recovery (action)      2/5 clear-cut correct   p=0.30-0.60   distinct answers=2/8
acceptance (plausible) 3/3 clear-cut correct   p=0.11-0.88
```

**The operative rule: temperature calibration sharpens ordering that is already right; it cannot make a wrong answer right.**

- **`review` specialty is the best question in the system** — payments→payments 0.67, security→security 0.70, data→data 0.62, general→general 0.63. Right ordering, low confidence. That is exactly the shape calibration fixes, so it is the kind most likely to produce a qualified bucket. Nothing in the original design predicted this.
- **`recovery` is near-degenerate** — "stop" for seven of eight states, including a first-attempt test failure. Three interventions were measured and none moved it: state as dict vs prose vs blocker-text-only (2/5 each), collapsing the five options to three or two (at two options it returns the same label at ~0.60 for *every* state), and replacing the 5-way choice with binary recognition questions. [Full table](https://github.com/hathbanger/orc/pull/24#issuecomment-5767797998).
  Why: whether a retry fixes `test failed: 3 assertions` depends on *why* it failed, and the blocker text does not say. A flaky assertion and a fatal import error read identically. No encoding recovers information that is not in the input. **Practical consequence: do not have anyone spend weeks labeling recovery decisions. It cannot pass the qualification gate.**
- **The generalizable rule:** ask only questions whose answer is determined by the input you pass. Review specialty is fully determined by the diff description; acceptance plausibility by the task and summary; recovery's action is not.

---

## 4. Three decisions that need you, not an agent

**a) #22 — close it?** It is a draft. It wires the two nouls that are currently inferred and discarded. I recommend closing: `needs_review` is near-constant (a one-line typo fix scores 0.64, a 50M-row migration 0.69), so gating the specialty on it makes the best question in the system **unreachable** — a regression against `main`. And `needs_clarification` is noise: the maximally-specified request scores 0.27, *higher* than one that explicitly lists its open questions (0.18). [Numbers](https://github.com/hathbanger/orc/pull/22#issuecomment-5767608628). Both nouls are already recorded in every decision record, so they stay available to label and re-evaluate after a fine-tune without this PR.

**b) The group trial.** There is a setup-and-labeling guide written for the 3–6 people who would run this: https://claude.ai/artifact/ErmLkV2yVCm7cdj4J8vLqf — setup, what to look at after each run, how to label with evidence, and the weekly export/calibrate loop. It is the human-readable version; the same material lives in `FUSION_DECISIONS.md`. The realistic floor to qualify one decision kind is **~100 labeled workflows per kind** (≥20 train groups, ≥20 confident held-out groups, ≥95% selective accuracy, ~20% of groups held out). Worth setting expectations with that number up front.

**c) Telemetry is now on by default with no token and no configuration (#27).** A shared write secret handed out of band meant nobody reported; ingest now takes no credential, the endpoint ships in the defaults, and the first send prints one line saying what is sent and that `FUSION_TELEMETRY=0` stops it. The token survives only as the *read* key for `GET /v1/summary`, so aggregate usage is not public — Alec has it, and contributors never need it. Nothing to hand out; the decision left is whether you are happy with on-by-default in a public repo.

## 5. Pointers

| thing | where |
|---|---|
| Combined, verified tree | `402goose/orc:mergeable-stack` |
| Stale — ignore | `stack-plus-telemetry`, `fusion-integration-stack` |
| Decision design and the learning loop | `FUSION_DECISIONS.md` |
| Why each change exists, with the measurements | `FUSION_RESEARCH.md` |
| Check the checkpoint yourself | `~/.local/share/orc/laya/bin/python test/laya_benchmark.py` |
| Acceptance question on six authored states | `test/laya_smoke.py --acceptance` |
| Shadow changes nothing, asserted | `make dogfood-paired` |
| Group trial guide | https://claude.ai/artifact/ErmLkV2yVCm7cdj4J8vLqf |

## 6. If an agent picks this up

Read `FUSION_DECISIONS.md` and `FUSION_RESEARCH.md` first — the second one records what was tried and failed, which is most of the value. Then §1 here.

Two habits that earned their keep and are worth keeping:

1. **Measure before encoding a rule.** Three separate design decisions in this batch were taken from plausible-sounding patterns and were wrong on the real model: the acceptance question wording (twice) and the paired-noul agreement rule (#22). Each was caught by running authored cases against the actual checkpoint. `test/laya_benchmark.py` exists so the next person measures first.
2. **Integrate before proposing.** #25's status check was too broad and flagged `cache_hit` from #16 as drift — a false positive that only appeared once the branch met the others. Merging all open branches locally before opening a PR caught it. [What happened](https://github.com/hathbanger/orc/pull/25#issuecomment-5767872641).

Everything above was run on a real Laya install (setup 2m21s, 807 MB checkpoint, 160–1000 ms warm inference, 25–45 s cold start per CLI invocation) with real Codex dispatches, not fixtures.

**Comment by 402goose** [comment]
Update: telemetry changed shape since this was written. It is now on by default with **no token and no configuration** (#27) — the old model required handing every person a shared secret and having them hand-edit `.fusion.json`, which is not how telemetry works anywhere and meant nobody reported. Ingest takes no credential now; the token survives only as the read key for `/v1/summary`.

Two consequences worth a look during review: on-by-default in a public repo means a stranger who clones this starts reporting, so the first send announces itself once per machine and `FUSION_TELEMETRY=0` stops it while keeping local traces. And the test suite and both dogfood scripts now opt out explicitly — without that, `make test` would report from every machine that ran it.

§1's merge list is unchanged; #27 is independent and merges onto `main` on its own. §4c above is updated.

**Comment by 402goose** [comment]
All eleven PRs described here are merged (#14–#25), plus #27–#33 since. The merge order, the measurement notes, and the three decisions are all resolved or superseded — the shadow-mode collection fix (#20) landed, acceptance shipped as a fifth decision kind (#19), and recovery was left alone as the measurements recommended.

Closing as complete. Current state is in the README and FUSION_DECISIONS.md.
